frontend/components/site/SiteGitHubBranches.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { SITE, GITHUB_BRANCHES } from '../../propTypes';
+import LoadingIndicator from '../loadingIndicator';
+import GitHubRepoLink from '../GitHubRepoLink';
+import BranchViewLink from '../branchViewLink';
+import githubBranchActions from '../../actions/githubBranchActions';
+
+export class SiteGitHubBranches extends React.Component {
+  componentDidMount() {
+    githubBranchActions.fetchBranches(this.props.site);
+  }
+
+  render() {
+    const githubBranches = this.props.githubBranches;
+    const site = this.props.site;
+
+    if (!site || githubBranches.isLoading) {
+      return <LoadingIndicator />;
+    }
+
+    if (!githubBranches.data || !githubBranches.data.length) {
+      return (
+        <p>
+          No branches were found for this repository.
+          There may have been an error communicating with the GitHub API.
```

Comment:
Should we put something in here about private repos?

I remember seeing something about adding a recommendation about private repositories to our docs. If something like that happens, another idea may be to link to that here.

Reply:
Good question. I'm actually not sure what would happen with a private repo!

Reply:
The GH API renders a 404.

In the old branch table, we handled the error by displaying a message that said something along these lines with a note that a private repo could also be to blame.

Reply:
Ah, cool. I'll change the wording and logic to be closer to the previous table's handling of this case.